Resolver/TenantResolver.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancy Component.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Component\MultiTenancy\Resolver;
+
+use SWP\Component\MultiTenancy\Repository\TenantRepositoryInterface;
+
+/**
+ * TenantResolver resolves the tenant based on subdomain.
+ */
+class TenantResolver implements TenantResolverInterface
+{
+    /**
+     * @var string
+     */
+    private $domain;
+
+    /**
+     * @var TenantRepositoryInterface
+     */
+    private $tenantRepository;
+
+    /**
+     * Construct.
+     *
+     * @param string                    $domain
+     * @param TenantRepositoryInterface $tenantRepository
+     */
+    public function __construct($domain, TenantRepositoryInterface $tenantRepository)
+    {
+        $this->domain = $domain;
+        $this->tenantRepository = $tenantRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($host = null)
+    {
+        // fallback to default tenant
+        if (null === $host) {
+            $host = self::DEFAULT_TENANT;
+        }
+
+        $subdomain = $this->extractSubdomain($host);
+
+        return $this->tenantRepository->findBySubdomain($subdomain);
```

Comment:
should not make extra query when the tenant is default. It should return populated (with default tenant data) object of type Tenant